packages/p2p/src/stores/order-store.js

```diff
@@ -485,13 +484,11 @@ export default class OrderStore {
                         response.error.code === api_error_codes.INVALID_VERIFICATION_TOKEN ||
                         response.error.code === api_error_codes.EXCESSIVE_VERIFICATION_REQUESTS
                     ) {
-                        clearTimeout(wait);
-                        this.setVerificationLinkErrorMessage(response.error.message);
-                        const wait = setTimeout(() => this.setIsInvalidVerificationLinkModalOpen(true), 750);
+                        this.root_store.general_store.showModal({
+                            key: 'InvalidVerificationLinkModal',
+                            props: { error_message: response.error.message, order_id },
+                        });
                     } else if (response.error.code === api_error_codes.EXCESSIVE_VERIFICATION_FAILURES) {
-                        if (this.is_invalid_verification_link_modal_open) {
-                            this.setIsInvalidVerificationLinkModalOpen(false);
-                        }
```

Comment:
Is it possible to handle this by getting the modal key from modal manager, checking against it and only hiding the modal if it is the `InvalidVerificationLinkModal`?

Reply:
I believe there's no need to pass the modal key when hiding since we are showing and hiding modals one at a time.